infobserve/common/pool.py

```diff
@@ -29,3 +31,24 @@ def acquire(self):
             (asyncpg.connection.Connection): A Connection instance.
         """
         return self.pool.acquire()
+
+
+class RedisConnectionPool(metaclass=Singleton):
+    redis = None
+
+    async def init_redis_pool(self, redis):
+        """Initialize Redis connection pool.
+        Arguments:
+            redis (dict): A dict with kwargs to initialize the connection pool.
+        """
+        self.redis: aioredis.RedisPool = await aioredis.create_pool((redis["host"], redis["port"]))
+
+    async def acquire(self):
+        """
+        Returns:
+            (aioredis.)
```

Comment:
^

Reply:
It's not even required because the way aioredis connection pool works the context closes so you can't return a connection cause the moment the context closes the connection closes also very weird so deleting it.